### PR TITLE
Replace old stb_image 1.18 with stb_image 2.30 and stb_image_write 1.16

### LINKED ISSUE #13: crunch is known to not properly grok all png format variants
I don't have samples to provide at this time, but crunch is known to not properly grok all png format variants. So, while this bug is not fixed, it's better to convert `png` to another format (example: `tga`) before trying to convert them with `crunch`.

### PR DESCRIPTION
**Pull request by illwieckz**
Replace old stb_image with stb_image 2.30 and stb_image_write 1.16.
    
Pulled from https://github.com/nothings/stb
Commit `013ac3beddff3dbffafd5177e7972067cd2b5083`

Fixes #13:

- https://github.com/DaemonEngine/crunch/issues/13

### CODE REVIEW

**Comment by illwieckz** [comment]
One can compare from the commit before this branch the `example1/stb_image.h` file with the `crnlib/crn_image_utils.cpp` file which was just a copy of the same `stb_image.h` at the time plus some minor changes.

The big difference is that previously, the stb functions were declared in `namespace crnlib`, it si not anymore.

Other changes is that many minor fixes implemented on `crnlib/crn_image_utils.cpp` are now lost, but all of them are probably useless and has been likely implemented by upstream in newer `stb_image.h`. In fact, I managed to produce both Linux, MinGW, macOS and FreeBSD builds without problems.

I guess it means this can introduce a mismatch if crnlib is linked with another software having built another version of stb.

**Comment by illwieckz** [comment]
The update of stb_image will also help to make the examples less Windows-centric, as the very old stb_image was Windows-centric and the copy in `crnlib` received fixes before.

**Comment by illwieckz** [comment]
So, Appveyor says it also builds on MSVC. So it builds everywhere as I verified myself the others.

**Comment by DolceTriade** [comment]
Did you make any modifications to the upstream file?

**Comment by illwieckz** [comment]
Except the changes done in `stb_image: fixes problems reported by CodeQL` commit (which are preventing some integers to overflow as reported by CodeQL), the stb files are an unmodified a copy-paste from stb upstream. I plan to submit the patch to STB upstream. My purpose is that future STB updates would just copy-paste the STB upstream.

**Comment by illwieckz** [comment]
This reintroduces a `sprintf`.

They have a pending PR for it:

- https://github.com/nothings/stb/pull/1619

**Review by DolceTriade — APPROVED** [review]

File: example2/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,8 @@ set(EXAMPLE2_SRCS
   ${CMAKE_CURRENT_SOURCE_DIR}/example2.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/timer.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/timer.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/../inc/crnlib.h
```
Comment (slipher) [inline example2/CMakeLists.txt:11]: Headers of a dependent library don't really belong in the source list. Same goes for example3

Comment (illwieckz) [reply example2/CMakeLists.txt:11]: Isn't it what CMake use to know the source has to be rebuilt if the header changes?

Comment (slipher) [reply example2/CMakeLists.txt:11]: No. AFAIK the only use is to index it in IDE projects.

Comment (illwieckz) [reply example2/CMakeLists.txt:11]: Ah, wouldn't it be a good idea though? I see there are some VS projects there but they are likely unmaintained and will be out of date once this is merged.

Comment (slipher) [reply example2/CMakeLists.txt:11]: No. In the VS project setting, headers that are part of the Crunch library should be in the library's sources, not the example sources.

Comment (illwieckz) [reply example2/CMakeLists.txt]: OK I removed them.

**Pull request merged**